pytorch_lightning/utilities/device_parser.py

```diff
@@ -199,15 +219,36 @@ def _normalize_parse_gpu_input_to_list(gpus: Union[int, List[int], Tuple[int, ..
     if not gpus:  # gpus==0
         return None
     if gpus == -1:
-        return _get_all_available_gpus()
+        return _get_all_available_gpus(include_cuda=include_cuda, include_mps=include_mps)
 
     return list(range(gpus))
 
 
-def _get_all_available_gpus() -> List[int]:
+def _get_all_available_gpus(include_cuda: bool = False, include_mps: bool = False) -> List[int]:
+    """
+    Returns:
+        a list of all available gpus
+    """
+    cuda_gpus = _get_all_available_cuda_gpus() if include_cuda else []
+    mps_gpus = _get_all_available_mps_gpus() if include_mps else []
+    return cuda_gpus + mps_gpus
```

Comment:
On an M1 with an additional cuda gpu, the sum of these lists could have duplicate entries like [0, 0] for example. Shouldn't the two options be mutually exclusive?

Reply:
apple does not support cuda gpus and there simply are no drivers :)